components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2181,6 +2181,37 @@ public ServiceProvider getApplication(int applicationId) throws IdentityApplicat
         }
     }
 
+    @Override
+    public ServiceProvider getApplicationWithRequiredAttributes(int applicationId, List<String> requiredAttributes)
+            throws IdentityApplicationManagementException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        try {
+            // Load basic application data
+            ServiceProvider serviceProvider = getBasicApplicationData(applicationId, connection);
+            if (CollectionUtils.isNotEmpty(requiredAttributes)) {
+                List<ServiceProviderProperty> propertyList = getServicePropertiesBySpId(connection, applicationId);
+                for (String requiredAttribute : requiredAttributes) {
+                    if (requiredAttribute.equals("advancedConfigurations")) {
```

Comment:
use a constant

Reply:
also flip equals to have <constant>.equals(<variable>) to avoid any potential NPEs